### ISSUE TITLE
renderToString with {pretty: true} causes undesirable newlines with text nodes

### ISSUE THREAD

**Issue by stevenle** [preactjs/preact-render-to-string#273]
**Issue:**
When using `renderToString(vdom, {}, { pretty: true })`, text nodes that normally do not have whitespace between it and its siblings are rendered on a newline, which can create undesirable whitespace issues. 

**Minimal sample project with the issue:**
https://stackblitz.com/edit/js-imcswy?file=index.js

**Sample code:**

```javascript
/** @jsx h */
import { h } from 'preact';
import renderToString from 'preact-render-to-string';

function Text() {
  return (
    <p>
      Lorem ipsum <a>click me</a>. Dolor sit amet.
    </p>
  );
}

const html = renderToString(<Text />, {}, { pretty: true });
console.log(html);
```

**console.log output:**

```html
<p>
	Lorem ipsum 
	<a>click me</a>
	. Dolor sit amet.
</p>
```

**Proposed resolution:**

When there is no explicit whitespace between a text node and its sibling, the pretty renderer should keep the output on the same line. For the example above, I'd expect an output of:

```html
<p>
	Lorem ipsum 
	<a>click me</a>. Dolor sit amet.
</p>
```

![preact renderToString pretty issue](https://user-images.githubusercontent.com/387282/207942977-dc85fdf1-428b-45e4-b224-1e819a61b405.jpg)

**Comment by brandon-lb** [comment]
Yeah, same thing happened to me. The browser ended up interpreting some regular text with spaces around it.

**Comment by bvandrc** [comment]
Seconded!
